plugins/techdocs-react/src/alpha.ts

```diff
@@ -53,6 +57,19 @@ export const attachTechDocsAddonComponentData = <P>(
   techDocsAddon: ComponentType<P>,
   data: TechDocsAddonOptions,
 ) => {
-  attachComponentData(techDocsAddon, TECHDOCS_ADDONS_KEY, data);
-  attachComponentData(techDocsAddon, getDataKeyByName(data.name), true);
+  const element = React.createElement(techDocsAddon as ComponentType);
```

Comment:
I'm a little confused on the fix here, and apologies for my very little knowledge of how all this fits together.

Both `attachComponentData` and `getComponentData` will be removed when the legacy frontend system goes away, so is there actually a different fix here?

Should we be removing the requirement that this is actually set instead of setting it on New Frontend System methods?

Reply:
From what I understand, all extension factories are called at startup. In techdocs case, we're using these 2 [blueprints](https://github.com/backstage/backstage/blob/8ab9af844cea5c467e28cda36e8b7205af647fb9/plugins/techdocs/src/alpha.tsx#L145-L218) which both call `attachTechDocsAddonComponentData`. Since it's being ran twice, the techdocs addons page ends up throwing an error on the entity docs page. This prevents the `attachComponentData` from being attempted twice on the same addon.

https://github.com/user-attachments/assets/5e6bd1dd-b566-4f61-9692-0b46bcc0304c

I am unaware that `attachComponentData` is being removed. That would require a bit of reworking done to techdocs addons which our team doesn't have capacity for atm.